doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,159 @@
+C++ Plugin API for CPU compressors  {#data}
+============
+
+This document specifies a modern API for the in-memory C plugin API for
+lossless byte data compressors. It will not replace the 2D image
+compressors and CPU-GPU transfer plugins.
+
+## Requirements
+
+* Support all features used by current lossless byte compressors
+    * Lossless compression of binary blobs
+    * Multiple compressors per plugin DSO
+* Information provided for each compressor:
+    * Compression speed (relative to RLE compressor)
+    * Compression ratio (relative to uncompressed data)
+* Compressors may produce multiple output chunks
+* Compressor engines are instantiated to allow reuse of internal state
+  and memory allocations
```

Comment:
Just by looking ad the Compressor API, it's not clear how internal state would be reused. What freedoms implementors have should be documented here, e.g., whether the buffers returned by compress are owned by the caller of the Compressor. Maybe this is implicit in the fact that Compressor::compress returns const&, but I overlooked it on the first reading. That makes me wonder how could the client code take ownership of the compressed data, otherwise It's necessary to keep the compressor alive while the compressed buffer is needed, furthermore, I need a separate compressor for each blob I want to compress concurrently.